docs/data/material/components/tree-view/GmailTreeView.tsx

```diff
@@ -24,69 +24,86 @@ declare module 'react' {
 
 type StyledTreeItemProps = TreeItemProps & {
   bgColor?: string;
+  bgColorForDarkMode?: string;
   color?: string;
+  colorForDarkMode?: string;
   labelIcon: React.ElementType<SvgIconProps>;
+  iconColorForDarkMode?: string;
   labelInfo?: string;
   labelText: string;
 };
 
-const StyledTreeItemRoot = styled(TreeItem)(({ theme }) => ({
-  color: theme.palette.text.secondary,
-  [`& .${treeItemClasses.content}`]: {
-    color: theme.palette.text.secondary,
-    borderTopRightRadius: theme.spacing(2),
-    borderBottomRightRadius: theme.spacing(2),
-    paddingRight: theme.spacing(1),
-    fontWeight: theme.typography.fontWeightMedium,
-    '&.Mui-expanded': {
-      fontWeight: theme.typography.fontWeightRegular,
-    },
-    '&:hover': {
-      backgroundColor: theme.palette.action.hover,
-    },
-    '&.Mui-focused, &.Mui-selected, &.Mui-selected.Mui-focused': {
-      backgroundColor: `var(--tree-view-bg-color, ${theme.palette.action.selected})`,
-      color: 'var(--tree-view-color)',
-    },
-    [`& .${treeItemClasses.label}`]: {
-      fontWeight: 'inherit',
-      color: 'inherit',
-    },
-  },
-  [`& .${treeItemClasses.group}`]: {
-    marginLeft: 0,
-    [`& .${treeItemClasses.content}`]: {
-      paddingLeft: theme.spacing(2),
-    },
-  },
-}));
-
 function StyledTreeItem(props: StyledTreeItemProps) {
+  const theme = useTheme();
   const {
     bgColor,
     color,
     labelIcon: LabelIcon,
     labelInfo,
     labelText,
+    colorForDarkMode,
+    bgColorForDarkMode,
+    iconColorForDarkMode,
     ...other
   } = props;
 
+  const styleProps = {
+    '--tree-view-color': theme.palette.mode !== 'dark' ? color : colorForDarkMode,
+    '--tree-view-bg-color':
+      theme.palette.mode !== 'dark' ? bgColor : bgColorForDarkMode,
+  };
+
   return (
-    <StyledTreeItemRoot
+    <TreeItem
       label={
-        <Box sx={{ display: 'flex', alignItems: 'center', p: 0.5, pr: 0 }}>
+        <Box
+          sx={{
+            display: 'flex',
+            alignItems: 'center',
+            p: 0.5,
+            pr: 0,
+            '& .MuiTypography-body2': {
+              fontWeight: 'inherit',
+              flexGrow: 1,
+            },
+          }}
+        >
           <Box component={LabelIcon} color="inherit" sx={{ mr: 1 }} />
-          <Typography variant="body2" sx={{ fontWeight: 'inherit', flexGrow: 1 }}>
-            {labelText}
-          </Typography>
-          <Typography variant="caption" color="inherit">
+          <Typography variant="body2">{labelText}</Typography>
+          <Typography variant="caption" color="inherit" sx={{ ml: 'auto' }}>
```

Comment:
Not needed. To keep it same as before:
```suggestion
          <Typography variant="caption" color="inherit">
```